pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -886,6 +888,35 @@ public void redeliverUnacknowledgedMessages() {
     }
 
     @Override
+    public void redeliverUnacknowledgedMessages(List<MessageIdImpl> messageIds) {
+        if (!conf.isPerMessageRedeliverUnacknowledged()
+                || messageIds.size() > conf.getMaxRedeliverUnacknowledgedMessagesBatch()) {
+            redeliverUnacknowledgedMessages();
+            return;
+        }
+        ClientCnx cnx = cnx();
+        if (isConnected() && cnx.getRemoteEndpointProtocolVersion() >= ProtocolVersion.v2.getNumber()) {
+            MessageIdData.Builder builder = MessageIdData.newBuilder();
```

Comment:
Right now `Commands` class cannot receive `MessageId` because `pulsar-common` doesn't and shouldn't have dependency on anything else.

How would you create a builder for this?

I've checked if I could find a similar case in `Commands` but couldn't find one